source/_integrations/ecowitt.markdown

```diff
@@ -21,13 +21,19 @@ ha_integration_type: integration
 
 ## Ecowitt Weather Station configuration
 
-The following steps must be performed to set up this integration. For security reason, use the token path that you get from the Home Assistant config flow.
+The following steps must be performed to set up this integration. For security reasons, use the token path that you get from the Home Assistant config flow.
+
+<div class='note'>
+
+If you have taken additional steps to enable HTTPS (secure) access to your Home Assistant instance, the Ecowitt integration will require extra configuration.
+
+Ecowitt devices do not support sending data to HTTPS URLs. A solution exists using the NGINX TLS Proxy Add-on to support HTTP and HTTPS at the same time (for more information read this [community discussion](https://community.home-assistant.io/t/nginx-tls-proxy-add-on-config-to-allow-simultaneous-https-and-http-access-required-by-ecowitt-integration/589276)).
```

Comment:
I don't think it is right to add documentation to third-party resources (in this case the community forums).

Reply:
That's not a hill I plan on dying on, who makes the call? If it's you, can you suggest an alternative to my following problem:

I only opened this PR because the existing documentation wasn't obvious, and it wasn't until I found that exact community discussion. It's also worth pointing out that the docs _already_ recommended using the named third-party extension (without further guidance) which was really unhelpful.

This issue (https://github.com/home-assistant/core/issues/93000) also helped me out a ton, but I figured a link to the community discussion would be better than a core, open, GitHub issue. Maybe that's better?

Lastly, there's not a single solution or official guidance on how to deal with this, so a discussion link opens up to alternatives and potentially long term guidance.

If we/you decide to remove the community link, can you help find a better way to document this exact problem and solution for Ecowitt that can remain part of this central notes? At the very least, an alternate form of guidance to unsuspecting consumers of this integration?

Reply:
> That's not a hill I plan on dying on, who makes the call? 

Yes.

>  can you suggest an alternative to my following problem:

Sorry, but this ain't a support thread.

> Lastly, there's not a single solution or official guidance on how to deal with this

It is not in our scope. If you want to use reverse proxies, that is up to you.

> If we/you decide to remove the community link, can you help find a better way to document this exact problem and solution for Ecowitt that can remain part of this central notes? At the very least, an alternate form of guidance to unsuspecting consumers of this integration?

I think it is enough to state that this won't work if Home Assistant itself isn't available on `http`.

Reply:
> Sorry, but this ain't a support thread.

Apologies, to be clear, not asking for "product support" but asking an open source maintainer, for documentation guidance, about real friction points that I'm trying to resolve for others; for no compensation other than for my own interests and time. I think it would help to be a bit more empathetic.

> I think it is enough to state that this won't work if Home Assistant itself isn't available on http.

I won't update the docs to me even more unhelpful by eliminating the existing mention of a third part integration, but can see your view point while disagreeing. Again, not my hill.

Feel free to close this PR if there's no interest in bettering the documentation for any Home Assistant users trying to add Ecowitt securely to their smart home.

Reply:
> I won't update the docs 

If you don't want to work on this PR, then I suggest closing it.

../Frenck

Reply:
The intent of the PR is to "Improve ecowitt docs around HTTP requirements" and the commit currently does that (IMO). I'm happy to work on it and revise the PR in ways that helps (impacted / running HTTPS) folks have a clear understanding of the action they need to take.

This is because running HTTPS is a best practice _and_ recommended by Home Assistant to secure it.

However, your current suggestion is to revise the entire PR to state, 

> "this won't work if Home Assistant itself isn't available on HTTP."

Leaving the docs in a worse state than they are today because it'd remove the existing tip for a solution. It also abandons anyone running Home Assistant on HTTPS (as they should be). That's not to mention the aforementioned issue. It's additionally a real problem for anyone trying to send data to Nabu Casa (financially supporting Home Assistant); as they default to HTTPS.

I have hopes that we can find a common ground to improve the docs for HTTPS users, so I'm reluctant to close the PR. But I fully respect your decision to do so; if there's no chance in achieving the goals set out to be resolved in this PR.